packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -526,85 +511,360 @@ export const Tabs = (props: TabsProps) => {
     }
   };
 
-  const createTab = React.useCallback(
-    (label: React.ReactNode, index: number) => {
-      const onClick = () => {
-        setFocusedIndex(index);
-        onTabClick(index);
-      };
-      return (
-        <li key={index}>
-          {!React.isValidElement(label) ? (
-            <Tab
-              label={label}
-              className={cx({
-                'iui-active': index === currentActiveIndex,
-              })}
-              tabIndex={index === currentActiveIndex ? 0 : -1}
-              onClick={onClick}
-              aria-selected={index === currentActiveIndex}
-            />
-          ) : (
-            React.cloneElement(label as JSX.Element, {
-              active: index === currentActiveIndex,
-              'aria-selected': index === currentActiveIndex,
-              tabIndex: index === currentActiveIndex ? 0 : -1,
-              onClick: (args: unknown) => {
-                onClick();
-                label.props.onClick?.(args);
-              },
-            })
-          )}
-        </li>
-      );
-    },
-    [currentActiveIndex, onTabClick],
+  return (
+    <Box
+      as='ul'
+      className={cx(
+        'iui-tabs',
+        `iui-${type}`,
+        {
+          'iui-green': color === 'green',
+          'iui-animated': type !== 'default' && isClient,
+          'iui-not-animated': type !== 'default' && !isClient,
+          'iui-large': hasSublabel,
```

Comment:
Didn't know where to leave the comment, but our method for getting `hasSublabel` is also a hack that needs to be addressed in this PR

Reply:
This hack has been updated. Adding a `Tab.Description` subcomponent changes the value of `hasSublabel` if the tab type is not a pill.